datafusion/core/src/physical_optimizer/sort_pushdown.rs

```diff
@@ -63,7 +63,13 @@ pub fn assign_initial_requirements(node: &mut SortPushDown) {
     for (child, requirement) in node.children.iter_mut().zip(reqs) {
         child.data = ParentRequirements {
             ordering_requirement: requirement,
-            fetch: None,
+            // If the parent has a fetch value, assign it to the children
+            // Or use the fetch value of the child.
+            fetch: if let Some(fetch) = child.plan.fetch() {
+                Some(fetch)
+            } else {
+                None
+            },
```

Comment:
Isn't this the same as this?

```suggestion
            fetch: child.plan.fetch().clone(),
```

Reply:
yes